sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -140,26 +140,36 @@
   public Span startSpan() {
     SpanContext parentContext = parent(parentType, parent, remoteParent);
     TraceId traceId;
+    SpanContext spanContext;
     SpanId spanId = idsGenerator.generateSpanId();
+    Decision samplingDecision;
     Tracestate tracestate = Tracestate.getDefault();
     if (parentContext == null || !parentContext.isValid()) {
       // New root span.
       traceId = idsGenerator.generateTraceId();
       // This is a root span so no remote or local parent.
       parentContext = null;
+      samplingDecision =
```

Comment:
this can be pulled up above the if block, since it's the same in both branches.

Reply:
we need the `traceId` for this which is created separately

Reply:
ah, I see. you are correct.